arch/arm/core/cortex_m/tz/CMakeLists.txt

```diff
@@ -27,11 +27,6 @@ if(CONFIG_ARM_FIRMWARE_HAS_SECURE_ENTRY_FUNCS)
     )
 endif()
 
-# Link the entry veneers library file with the Non-Secure Firmware that needs it.
-zephyr_link_libraries_ifdef(CONFIG_ARM_FIRMWARE_USES_SECURE_ENTRY_FUNCS
-  ${CMAKE_BINARY_DIR}/${CONFIG_ARM_ENTRY_VENEERS_LIB_NAME}
-  )
-
```

Comment:
Not sure what your intentions are with this temphack.

If we disregard the `CONFIG_ARM_FIRMWARE_USES_SECURE_ENTRY_FUNCS` and do not set it, this code will not be activated.

Instead we could do similar to what was earlier prototyped here:
https://github.com/tejlmand/fw-nrfconnect-nrf/blob/da2af5e2ca77d12f4f20295fbfeac9ff47c36c04/subsys/spm/Kconfig#L13
https://github.com/tejlmand/fw-nrfconnect-nrf/blob/39007f4caf2e610ca8e745a27e3aa01288745e41/cmake/multi_image.cmake#L145-L151

of course with the addition of allowing the user to not link with `veneers` if desired.
That way we don't need a temphack locally.

Reply:
I can't recall my plans for how to get rid of the temp hack, I will look into this.

I don't want to unset CONFIG_ARM_FIRMWARE_USES_SECURE_ENTRY_FUNCS because that would be incorrect, the image does use secure entry functions.

Reply:
I got rid of the temp hack now.

The default string is now changed based on whether the image is using SPM or is SPM:

https://github.com/NordicPlayground/fw-nrfconnect-nrf/pull/1508/files#diff-4098c32be5d8477048a5e432e1d5066cR225

Reply:
> I got rid of the temp hack now.

Great.